### PR TITLE
improvement(new event): create system status events

### PR DESCRIPTION
**Pull request by juliayakovlev**
Task: https://trello.com/c/EqQGzQVJ/3923-create-system-events
Create set of events regarding system status with severity=WARNING:
1. System startup
2. System reboot
3. System poweroff

Current implementation of power-off event change from DatabaseLogEvent since it is not relate to
DB events at all. Refactor this.

Events example:
```
t:2021-10-04 06:12:10,460 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-10-04 06:12:08.157 <2021-09-23 12:41:50.000>: (InstanceStatusEvent Severity.WARNING) period_type=one-time event_id=370c3d62-7381-473b-b4e2-85d6a51eb074: type=STARTUP regex=kernel: Linux version line_number=2 node=Node longevity-10gb-3h-system-e-db-node-ebb700a4-3 [13.53.182.150 | 10.0.2.227] (seed: False)
t:2021-10-04 06:12:10,460 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-09-23T12:41:50+00:00  ip-10-0-2-227 !NOTICE  | kernel: Linux version 5.4.0-1035-aws (buildd@lgw01-amd64-057) (gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)) #37-Ubuntu SMP Wed Jan 6 21:01:57 UTC 2021 (Ubuntu 5.4.0-1035.37-aws 5.4.78)

t:2021-10-04 06:12:10,462 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-10-04 06:12:09.630 <2021-09-23 12:45:20.000>: (InstanceStatusEvent Severity.WARNING) period_type=one-time event_id=adb5b13a-86f3-4560-b086-9ead2bcedb75: type=REBOOT regex=Stopped target Host and Network Name Lookups line_number=3214 node=Node longevity-10gb-3h-system-e-db-node-ebb700a4-3 [13.53.182.150 | 10.0.2.227] (seed: False)
t:2021-10-04 06:12:10,462 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-09-23T12:45:20+00:00  ip-10-0-2-227 !INFO    | systemd[1]: Stopped target Host and Network Name Lookups.

t:2021-10-04 06:12:10,464 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-10-04 06:12:09.712 <2021-09-23 12:45:28.000>: (InstanceStatusEvent Severity.WARNING) period_type=one-time event_id=ca3fd891-4a4a-4369-a51d-3c270aca24be: type=POWER_OFF regex=Reached target Power-Off line_number=3475 node=Node longevity-10gb-3h-system-e-db-node-ebb700a4-3 [13.53.182.150 | 10.0.2.227] (seed: False)
t:2021-10-04 06:12:10,464 f:file_logger.py  l:83   c:sdcm.sct_events.file_logger p:INFO  > 2021-09-23T12:45:28+00:00  ip-10-0-2-227 !INFO    | systemd[1]: Reached target Power-Off.
```

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: defaults/severities.yaml [hunk 1]
```diff
@@ -98,3 +98,6 @@ ScyllaServerStatusEvent: NORMAL
 BootstrapEvent: NORMAL
 RepairEvent: NORMAL
 JMXServiceEvent: NORMAL
+InstanceStatusEvent.STARTUP: WARNING
```
Comment (enaydanov) [inline defaults/severities.yaml:101]: Need to remove `DatabaseLogEvent.POWER_OFF: WARNING` line in this file

File: sdcm/sct_events/system.py [hunk 2]
```diff
@@ -222,3 +223,22 @@ def __reduce__(self):
...
+
+
+InstanceStatusEvent.add_subevent_type("STARTUP", severity=Severity.WARNING, regex="kernel: Linux version")
+InstanceStatusEvent.add_subevent_type("REBOOT", severity=Severity.WARNING,
+                                      regex="Stopped target Host and Network Name Lookups")
+InstanceStatusEvent.add_subevent_type("POWER_OFF", severity=Severity.WARNING, regex="Reached target Power-Off")
+
+INSTANCE_STATUS_EVENTS = (InstanceStatusEvent.STARTUP(),
+                          InstanceStatusEvent.REBOOT(),
+                          InstanceStatusEvent.POWER_OFF())
```
Comment (enaydanov) [inline sdcm/sct_events/system.py:241]: Please, use same style for the tuple items as in other files (see `database.py` for example)

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: defaults/severities.yaml (hunk 1 above)
Comment (juliayakovlev) [reply defaults/severities.yaml:100]: removed, thanks

File: sdcm/sct_events/system.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/sct_events/system.py]: changed

**Review by enaydanov — APPROVED** [review]

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1427,10 +1427,10 @@ def _read_system_log_and_publish_events(self,
                             item.period_func(match=event_match)
 
                     # for each line use all regexes to match, and if found send an event
-                    for pattern, event in SYSTEM_ERROR_EVENTS_PATTERNS:
+                    for pattern, event in SYSTEM_ERROR_EVENTS_PATTERNS + INSTANCE_STATUS_EVENTS_PATTERNS:
```
Comment (KnifeyMoloko) [inline sdcm/cluster.py:1430]: We can extract this to a variable somewhere outside of the loop, to avoid repeating this operation on every iteration.

**Review by KnifeyMoloko — CHANGES_REQUESTED** [review]
One small request for a change, otherwise looks good :)

**Review by KnifeyMoloko — APPROVED** [review]

File: sdcm/cluster.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/cluster.py]: done

**Pull request merged**